### DISCUSSION TITLE
Add support for recursive yield

### DISCUSSION THREAD

**Discussion by dgolub** [dotnet/csharplang#5705] [Language Ideas] [answered]
It would be great if the `yield` keyword could be used recursively to combine multiple `IEnumerable` objects or add elements to such an object.  For example, if we wrote
```
IEnumerable<string> A() {
    yield return "foo";
    yield return "bar";
}

IEnumerable<string> B() {
    yield return "x";
    yield return "y";
}

IEnumerable<string> C() {
    yield return "a";
    yield A();
    yield B();
    yield return "b";
}
```

Then, the function `C` would produce an `IEnumerable` yielding the elements `"a"`, `"foo"`, `"bar"`, `"x"`, `"y"`, and `"b"`.

**Comment by svick** [comment, marked as answer]
This is a duplicate of https://github.com/dotnet/csharplang/discussions/5303.
